modules/configuration/ajax/updateProject.php

```diff
@@ -50,6 +50,7 @@
     // Update Project fields
     $project = \Project::getProjectFromID($projectID);
     $project->updateName($projectName);
+    $project->updateAlias($projectAlias);
```

Comment:
It would be good to add a back-end check to the front-end check that Alias is > 4.

An InvalidArgumentException should be thrown by the Project class when Alias > 4 characters.

Currently this is done only in `setAlias()`, but not in the constructor or in `updateAlias()`.

I'd suggest adding a `_validateAlias(): void` function that throws the Exception based on the `_projectAlias` variable and then calling that function in set, update, and the constructor.

Reply:
Come to think of it this would be a good use case for an Alias value object that can also be used by Site. 

We can make a separate issue for this and punt to the next release if you prefer.

Reply:
Shouldnt we be then using the set functions and then updateProject, instead of separate update functions for each value ?

Reply:
and for the alias object i would punt it because we dont have the same criteria for a psc alias, a mri alias and a project alias, we would need to standardise first

Reply:
Created #5565 for adding a value object.

Reply:
>Shouldnt we be then using the set functions and then updateProject, instead of separate update functions for each value ?

That would work too, just depends on how we want the class to work. I don't mind either way

Reply:
I'll add the validation still but we'll rethink the logic later

Reply:
I made it pass through set which allowed me to remove the `??`